fe/fe-core/src/main/java/com/starrocks/planner/OlapScanNode.java

```diff
@@ -481,12 +484,21 @@ public void addScanRangeLocations(Partition partition,
                 replicas = allQueryableReplicas;
             }
 
+            if (!hintsReplicaIds.isEmpty()) {
+                replicas.removeIf(replica -> hintsReplicaIds.contains(replica.getId()));
```

Comment:
Why remove the replica that satisfy `hintsReplicaIds `?

Reply:
+1

Reply:
fixed